wordpress-mobile/WordPress-Android
# [Tooling] Fix hotfix lanes

Part of the paaHJt-1WQ-p2 project.

## To test

 - Checkout this branch, run `bundle install`
 - Run `bundle exec fastlane new_hotfix_release version_name:17.7.1 version_code:1077 app:wordpress`
 - Verify that it cut a `release/17.7.1` branch from the `17.7` tag
 - Verify that it bumped the `wordpress.*` versions in `version.properties` (but not the zalpha ones, we don't do alpha for hotfixes)
 - cherry-pick the 2 commits from this PR on top of the `release/17.7.1` branch. This is because we are gonna test the updated finalize lanes, and need to be on the hotfix release branch to test them.
 - [Open the CircleCI page for WPAndroid](https://circleci.com/gh/wordpress-mobile/WordPress-Android) in advance, to be ready to cancel the release build
 - Run `bundle exec fastlane finalize_release app:wordpress`. You should get an error telling you that you should use `finalize_hotfix_release` for hotfix branches instead
 - Run `bundle exec fastlane finalize_hotfix_release app:wordpress` and verify that it triggered a release build, and that is is building it from the release/17.7.1 hotfix branch. Cancel the CI build immediately (NB: `1077` is a version code that already exists so in case you fail to cancel it, the build should fail to upload to playstore anyway)
 - Delete the `release/17.7.1` branch (from both local and remote)

## PR Dependency ("Not Ready for Merge" status)

This PR depends on https://github.com/wordpress-mobile/release-toolkit/pull/288 ; once this gets merged, we need to do a new release of the toolkit and update the reference in `Pluginfile` before merging this.
